controllers/api/userRoutes/userRoutes.js

```
@@ -29,100 +58,212 @@ router.get('/', auth, isAdmin, async (req, res) => {
 });
 
-router.get('/:id', isAdmin, async (req, res) => {
-  const { id } = req.user;
+router.get('/search', auth, isAdmin, async (req, res) => {
+  //users can be searched by one or more of the following fields. searches are case-insensitive
+  const { email, first_name, last_name } = req.query;
+
   try {
-    const user = await Users.findByPk(
-      id, 
-      {
-        attributes: ['id', 'first_name', 'last_name', 'email'],
-      },
-    );
-    if (!user) {
-      return res.status(404).json({ message: 'User not found' });
+    // Initialize the base `where` clause
+    let whereClause = {};
+    // Restrict data visibility based on user role
+
+    if (req.user.roleId === 2) {
+      // Role 2: Can only view users within the same organization
+      whereClause.organization_id = req.user.organization_id;
+    } else if (req.user.roleId !== 3) {
+      // Role not recognized, deny access (should not happen due to `isAdmin` middleware)
+      return res.status(403).json({ message: 'You do not have permission to search for users' });
+    }
+
+    // Add case-insensitive search filters if provided
+    if (email) {
+      whereClause.email = {
+        [Op.like]: `%${email.toLowerCase()}%`,  // Ensure email in the db is compared case-insensitively
+      };
     }
+    if (first_name) {
+      whereClause.first_name = {
+        [Op.like]: `%${first_name.toLowerCase()}%`,
+      };
+    }
+    if (last_name) {
+      whereClause.last_name = {
+        [Op.like]: `%${last_name.toLowerCase()}%`,
+      };
+    }
+
+    // Fetch the users with filters
+    const users = await Users.findAll({
+      where: whereClause,  // Use the correct `where` object
+      attributes: ['id', 'first_name', 'last_name', 'email'],
+      include: [
+        {
+          model: Roles,
+          as: 'role',
+          attributes: ['name'],
+        },
+      ],
+    });
+
+    res.status(200).json(users);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
```

don't forget to console log the error here as well, else it gets swallowed